src/cephadm/cephadm

```diff
@@ -2498,6 +2498,8 @@ def get_container(ctx: CephadmContext,
         # So the container can modprobe iscsi_target_mod and have write perms
         # to configfs we need to make this a privileged container.
         privileged = True
+        # --pids-limit 0 to allow more than 2048 threads
+        container_args.append('--pids-limit=0')
```

Comment:
So we are consciously ignoring `TasksMax=infinity` from `systemd/ceph*@.service.in` templates?

Reply:
unfortunately the `cephadm` binary has not yet access to the templates there. That depends on #41855 . See 

https://github.com/ceph/ceph/blob/f0b79b3a48e78a4b47c5c23c3991728cee01887a/src/cephadm/cephadm#L3094-L3146

for the current template.

Reply:
Since all those templates override `TasksMax`, perhaps `TasksMax=infinity` could be added to this template in the interim?

Reply:
Just wondering because `--pids-limit` seems to be treated differently between `podman` and `docker`:
```
$ podman run --help | grep pids-limit
      --pids-limit int                           Tune container pids limit (set 0 for unlimited, -1 for server defaults) (default 2048)

$ docker run --help | grep pids-limit
      --pids-limit int                 Tune container pids limit (set -1 for unlimited)
```

Reply:
LimitNOFILE etc is there already:

https://github.com/ceph/ceph/blob/f0b79b3a48e78a4b47c5c23c3991728cee01887a/src/cephadm/cephadm#L3122-L3123

So this must come from somewhere else. But you're right, we should probably see, if we can adopt other settings as well.

Reply:
Ah, I missed the shared section.  Gihub quotes can be deceiving...

So why `TasksMax=infinity` isn't there in the shared section?  Is there a particular reason it was omitted?

Reply:
pushed. Does this work for you? Yes, we need to get the service files in sync definitely.